frontend/components/sections/BitTableSection.tsx

```diff
@@ -111,55 +101,121 @@ function Bit({ type, driverSizeColor, sizes }: GroupedBit) {
+                        color={`${driverSizeColors[driverSize]}.700`}
                         fontSize="xs"
                         fontFamily="mono"
                      >
-                        {size}
+                        {driverSize}
                      </Text>
                   </Flex>
                ))}
             </Flex>
          </Flex>
+         <Flex flexGrow={1} alignItems="center" gap="3" p="3">
+            <Flex gap="1">
+               {bitCount && (
+                  <>
+                     <Text flexGrow={1}>{bitCount}</Text>
+                     <Text color="gray.400">×</Text>
+                  </>
+               )}
+               <Text>{type.name}</Text>
+            </Flex>
+
+            <Flex flexGrow="1" flexDirection="column" gap={1}>
+               {Object.entries(sizes).map(([driverSize, sizesForDriver]) => {
+                  return sizesForDriver.length > 0 ? (
+                     <Flex
+                        key={driverSize}
+                        flexWrap="wrap"
+                        gap={1}
+                        justifyContent="flex-end"
+                     >
+                        {sizesForDriver.map((size) => (
+                           <Flex
+                              key={size}
+                              px="1.5"
+                              h="4"
+                              gap="1"
+                              bg="gray.100"
+                              borderRadius="full"
+                              borderColor="gray.200"
+                              borderWidth={1}
+                              alignItems="center"
+                              justify="center"
+                              minW="9"
+                           >
+                              <Text
+                                 as="span"
+                                 color="gray.800"
+                                 fontSize="xs"
+                                 fontFamily="mono"
+                              >
+                                 {size}
+                              </Text>
+                              <Box
+                                 w="1.5"
+                                 h="1.5"
+                                 bg={`${driverSizeColors[driverSize]}.500`}
+                                 borderRadius="full"
+                              ></Box>
+                           </Flex>
+                        ))}
+                     </Flex>
+                  ) : null;
+               })}
+            </Flex>
+         </Flex>
       </Flex>
    );
 }
+
+const groupScrewdriverBits = (bits: ScrewdriverBit[]): GroupedBit[] => {
+   const groups = Object.values(groupBy(bits, (bit) => bit.type.name));
+   return groups.map((group) => {
+      const sizes: Record<string, string[]> = mapValues(
+         groupBy(group, (bit) => bit.type.driverSize),
+         (bits) => bits.map((bit) => bit.size).filter(isNotNull)
```

Comment:
To have duplicated bit one have to create a duplicated `ScrewdriverBit` in Strapi and also insert it twice in the `BitTableSection`.
It is not impossible but in case it happen I'd prefer the error to pop up so that the duplicate data can be identified and the strapi data source can be cleaned up.

Reply:
Works for me for now. I subscribe to the "make errors unrepresentable" course; I'd suggest we consider whether there's a better data model which avoids this possibility. If not, I might not worry too hard about it, because I'm not sure that we want invisible restrictions on what the folks editing Strapi can do.